ee/clickhouse/models/test/test_dead_letter_queue.py

```diff
@@ -0,0 +1,117 @@
+import json
+from datetime import datetime
+from time import sleep
+from uuid import UUID, uuid4
+
+from kafka import KafkaProducer
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.sql.dead_letter_queue import DEAD_LETTER_QUEUE_TABLE, INSERT_DEAD_LETTER_QUEUE_EVENT_SQL
+from ee.clickhouse.util import ClickhouseTestMixin, delay_until_clickhouse_consumes_from_kafka
+from ee.kafka_client.topics import KAFKA_DEAD_LETTER_QUEUE
+from posthog.settings import KAFKA_HOSTS
+from posthog.test.base import BaseTest
+
+TEST_EVENT_RAW_PAYLOAD = json.dumps(
+    {"event": "some event", "properties": {"distinct_id": 2, "token": "invalid token",},}
+)
+
+CREATED_AT = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
+ERROR_TIMESTAMP = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
+NOW = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
+
+
+TEST_DATA = {
+    "id": str(uuid4()),
+    "event_uuid": str(uuid4()),
+    "event": "some event",
+    "properties": "{ a: 1 }",
+    "distinct_id": "some distinct id",
+    "team_id": 1,
+    "elements_chain": "",
+    "created_at": CREATED_AT,
+    "ip": "127.0.0.1",
+    "site_url": "https://myawesomewebsite.com",
+    "now": NOW,
+    "raw_payload": TEST_EVENT_RAW_PAYLOAD,
+    "error_timestamp": ERROR_TIMESTAMP,
+    "error_location": "plugin-server",
+    "error": "createPerson failed",
+}
+
+
+def reset_tables():
+    sync_execute("TRUNCATE TABLE events_dead_letter_queue")
+
+    # can't truncate table with kafka engine, reading from it will delete the rows
+    sync_execute("SELECT * FROM kafka_events_dead_letter_queue")
+
+
+class TestDeadLetterQueue(ClickhouseTestMixin, BaseTest):
+    def test_direct_table_insert(self):
+
+        sync_execute(
+            INSERT_DEAD_LETTER_QUEUE_EVENT_SQL, TEST_DATA,
+        )
+
+        dead_letter_queue_events = sync_execute("SELECT * FROM events_dead_letter_queue LIMIT 1")
+
+        dlq_event = dead_letter_queue_events[0]
+
+        self.assertEqual(type(dlq_event[0]), UUID)  # id
+        self.assertEqual(type(dlq_event[1]), UUID)  # event_uuid
+        self.assertEqual(dlq_event[2], "some event")  # event
```

Comment:
Instead of hardcoding the values from `TEST_DATA`, what about using `TEST_DATA['event']`?

Reply:
the idea here was to just be explicit in the assertion, as we tend to do with most of our tests. Happy to change though, don't feel strongly about this

Reply:
I didn't know we were doing this also in other places. I'm fine with that, I just thought the other way was maybe less copy paste.